### DISCUSSION TITLE
Support Weylus on linux

### DISCUSSION THREAD

**Discussion by bbb651** [ppy/osu#14311] [Ideas] [answered]
There's a program called [Weylus](https://github.com/H-M-H/Weylus) that allows you to use a device such as an iPad as a tablet for the computer, using a web browser connected to a web server on the host machine. On windows/mac it only emulates a mouse, but on linux it also has the ability to emulate a stylus (for example when using the apple pencil) and multi-touch, in all apps I tested that support a tablet it appears like one, here is how it looks in Krita's tablet tester:
![kritatablettester](https://user-images.githubusercontent.com/53972231/129479668-7c69d44b-bedb-4f30-aebf-a316169c2369.png)
Unfortunately it doesn't seem to work in osu!lazer, it doesn't show up in the settings as a tablet, and ingame it teleports the mouse to the bottom right corner and other weird behavior which makes it unusable. I'm not sure if the problem is on Weylus's end or osu!'s end (or if I messed up something while setting it up) but I feel like this could be nice to support this use case, it's a bit niche, but I think it could provide a really good experience, in other programs input is fast and precise.

**Comment by peppy** [comment, marked as answer]
Please turn off "high precision input" and it should work.

**Reply by bbb651** [reply]
Would be better if it worked as a tablet but this works really well when disabling mouse buttons, thanks!

**Reply by peppy** [reply]
You can try and request support over at OTD (https://github.com/OpenTabletDriver/OpenTabletDriver). This would be where it needs to be implemented in order to gain support as a tablet.
